crates/shared/Cargo.toml

```diff
@@ -45,6 +45,7 @@ prometheus = { workspace = true }
 prometheus-metric-storage = { workspace = true }
 rate-limit = { path = "../rate-limit" }
 reqwest = { workspace = true, features = ["cookies", "gzip", "json"] }
+rust_decimal = "1.35.0"
```

Comment:
This is something I wanted to do for some time. Rust decimal supports up to 28 decimal places of precision, and I believe we should be doing calculations with it rather than `f64` which has floating point/rounding imprecision.

Reply:
But we already have BigRational.

Reply:
`BigRational` is not as good as `rust_decimal`. `BigRational` keeps precision as long as the number can be relatively easily represented as a fraction, otherwise it is approximated. `BigRational` is slower due to its management of big numbers as fractions (basically `BigRational` works with fractions as floating point numbers). Because of that `BigRational` as dynamic precision.

Whereas, `rust_decimal` has always exact precision of 28 decimal points. Its only bad point is that it cannot handler super small numbers due to its 28 decimal point limitation, but within those 28 decimal points, the precision is exact.